system-addon/content-src/components/TopSites/TopSites.jsx

```diff
@@ -127,28 +125,19 @@ export class _TopSites extends React.PureComponent {
               <FormattedMessage id="edit_topsites_add_button" />
             </button>
           </div>
-          {showAddForm &&
-            <div className="edit-topsites">
-              <div className="modal-overlay" onClick={this.onFormClose} />
-              <div className="modal">
-                <TopSiteForm onClose={this.onFormClose} dispatch={this.props.dispatch} intl={this.props.intl} />
-              </div>
-            </div>
-          }
-          {showEditForm &&
-            <div className="edit-topsites">
-              <div className="modal-overlay" onClick={this.onFormClose} />
-              <div className="modal">
-                <TopSiteForm
-                  label={editSite.label || editSite.hostname || ""}
-                  url={editSite.url || ""}
-                  index={editIndex}
-                  editMode={true}
-                  onClose={this.onFormClose}
-                  dispatch={this.props.dispatch}
-                  intl={this.props.intl} />
-              </div>
+          {(showAddForm || editIndex >= 0) &&
+          <div className="edit-topsites">
+            <div className="modal-overlay" onClick={this.onFormClose} />
+            <div className="modal">
+              <TopSiteForm
```

Comment:
@rlr probably out of scope here, but the modal including `<TopSiteForm>` probably don't even need to be within `TopSites` anymore as we don't need to overlay one list over another?

Reply:
Yeah, this can probably be cleaned up more. Although `TopSites` is currently keeping track of the state that shows/hides this. Also, all that modal boilerplate should probably just go inside the component itself now too.

Reply:
Actually.. what's the `TopSites` component state vs redux `TopSites.editForm` state?

Reply:
yeah, that's to trigger edit from the `ContextMenu`. So we could switch to only using that and break out the `TopSiteForm` into somewhere else. Or keep it here and just get rid of the state.

Reply:
Let's just leave this inline modal for now and get rid of the `TopSites` component state to just use the redux `TopSites.editForm` state instead.